src/main/java/producer/RecordBatch.java

```diff
@@ -0,0 +1,62 @@
+package producer;
+
+import org.tinylog.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecordBatch {
+    private final int maxBatchSizeInBytes;
+    private int currBatchSize;
+    private int numRecords;
```

Comment:
I'm wondering if `numRecords` is needed, if we could call `batch.size()`

Reply:
true, probably a redundant var